src/NUnitEngine/nunit.engine/Services/AgentProcess.cs

```diff
@@ -0,0 +1,109 @@
+// ***********************************************************************
+// Copyright (c) Charlie Poole and TestCentric Engine contributors.
+// Licensed under the MIT License. See LICENSE.txt in root directory.
+// ***********************************************************************
+
+#if NETFRAMEWORK
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+using NUnit.Engine;
+using NUnit.Engine.Internal;
+
+namespace NUnit.Engine.Services
+{
+    public class AgentProcess : Process
```

Comment:
I strongly prefer composition over inheritance when layering domain-specific functionality on top of existing types.

Reply:
Me too! But there's a narrow band where either one may work well. In this case, I wanted the `AgentProcess` to actually __be__ a process. You'll notice that I also use composition. That is, the `AgentProcess` figures out where it's agent is located and does a few other things required to make the process work, while TestAgency sets up some other stuff, like monitoring when the process exits. I think the balance here is pretty reasonable, if I do say so myself. :smile:

Reply:
Niggle - could this class be internal? I know we don't have a good history of doing that for the engine, but I'd like to start. 🙂